Code/autopkglib/SparkleUpdateInfoProvider.py

```diff
@@ -181,59 +235,64 @@ def get_feed_data(self, url):
-                            item["version"] = filename.split(delimiter)[-1]
-                            break
-                # if we still found nothing, fail
-                if item["version"] is None:
-                    raise ProcessorError(
-                        "Can't extract version info from item in feed!"
-                    )
+                item["url"] = self.build_url(enclosure)
+                item["version"] = self.determine_version(enclosure, item["url"])
 
-                human_version = enclosure.get("{%s}shortVersionString" % xmlns)
+                human_version = enclosure.get("{%s}shortVersionString" % self.xmlns)
                 if human_version is not None:
                     item["human_version"] = human_version
-                min_version = item_elem.find("{%s}minimumSystemVersion" % xmlns)
+
+                min_version = item_elem.find("{%s}minimumSystemVersion" % self.xmlns)
                 if min_version is not None:
                     item["minimum_os_version"] = min_version.text
-                description_elem = item_elem.find("{%s}releaseNotesLink" % xmlns)
+
+                description_elem = item_elem.find("{%s}releaseNotesLink" % self.xmlns)
                 # Strip possible surrounding whitespace around description_url
                 # element text as we'll be passing this as an argument to a
                 # curl process
                 if description_elem is not None:
                     item["description_url"] = description_elem.text.strip()
+
                 if item_elem.find("description") is not None:
                     item["description_data"] = item_elem.find("description").text
                 versions.append(item)
 
         return versions
 
+    def handle_pkginfo(self, latest):
+        """Handles any keys we may have defined"""
+
+        pkginfo = {}
+        sparkle_pkginfo_keys = self.env.get("pkginfo_keys_to_copy_from_sparkle_feed")
+        if sparkle_pkginfo_keys:
+            for k in sparkle_pkginfo_keys:
+                if k not in SUPPORTED_ADDITIONAL_PKGINFO_KEYS:
+                    self.output(
+                        "Key %s isn't a supported key to copy from the "
+                        "Sparkle feed, ignoring it." % k
+                    )
+            # Format description
+            if "description" in sparkle_pkginfo_keys:
+                if "description_url" in latest.keys():
+                    description = self.fetch_content(latest["description_url"])
+                elif "description_data" in latest.keys():
+                    description = (
+                        "<html><body>" + latest["description_data"] + "</html></body>"
+                    )
+                else:
+                    description = ""
+                pkginfo["description"] = description.decode("UTF-8")
+
+            if "minimum_os_version" in sparkle_pkginfo_keys:
+                if latest.get("minimum_os_version") is not None:
```

Comment:
None is already not truth, so you can compress this:
```
if latest.get("minimum_os_version")
```

Yes, I know and see that this construction is everywhere, but it's something we'll have to change on our next revision.

Reply:
What if `minimum_os_version` is zero?

I think `is not None` construction is there for this purpose thus I did not change it in the refactoring